File: examples/node/sqs-create-queue.js
// Crea una cola SQS en Floci.
// Uso: node sqs-create-queue.js [nombre-de-cola]
const AWS = require('aws-sdk');

const sqs = new AWS.SQS({
  endpoint: 'http://localhost:4566',
  region: 'us-east-1',
  credentials: { accessKeyId: 'test', secretAccessKey: 'test' },
});

const queueName = process.argv[2] || `mi-cola-${Date.now()}`;

async function main() {
  const result = await sqs.createQueue({ QueueName: queueName }).promise();
  console.log('Cola creada:', queueName);
  console.log('URL:', result.QueueUrl);
}

main().catch((error) => {
  console.error('Error creando la cola:', error.message);
  process.exit(1);
});

File: examples/go/sqs_create_queue.go
// Crea una cola SQS en Floci.
// Uso: go run sqs_create_queue.go [nombre-de-cola]
package main

import (
	"context"
	"fmt"
	"log"
	"os"
	"time"

	"github.com/aws/aws-sdk-go-v2/aws"
	"github.com/aws/aws-sdk-go-v2/config"
	"github.com/aws/aws-sdk-go-v2/credentials"
	"github.com/aws/aws-sdk-go-v2/service/sqs"
)

func main() {
	ctx := context.Background()
	cfg, err := config.LoadDefaultConfig(ctx,
		config.WithRegion("us-east-1"),
		config.WithCredentialsProvider(credentials.NewStaticCredentialsProvider("test", "test", "")),
	)
	if err != nil {
		log.Fatalf("Error cargando configuracion: %v", err)
	}
	client := sqs.NewFromConfig(cfg, func(o *sqs.Options) {
		o.BaseEndpoint = aws.String("http://localhost:4566")
	})

	queueName := fmt.Sprintf("mi-cola-%d", time.Now().UnixMilli())
	if len(os.Args) > 1 {
		queueName = os.Args[1]
	}

	result, err := client.CreateQueue(ctx, &sqs.CreateQueueInput{QueueName: aws.String(queueName)})
	if err != nil {
		log.Fatalf("Error creando la cola: %v", err)
	}
	fmt.Println("Cola creada:", queueName)
	fmt.Println("URL:", aws.ToString(result.QueueUrl))
}
